src/GraphQL.Tests/Bugs/Bug1699InvalidEnum.cs

```diff
@@ -93,6 +93,9 @@ private void AssertQueryWithError(string query, string result, string message, i
         [Fact]
         public void Input_Enum_OverrideDefault() => AssertQuerySuccess("{ inputWithDefault(arg: SLEEPY) }", @"{ ""inputWithDefault"": ""Sleepy"" }");
 
+        [Fact]
+        public void Input_Enum_NullDefault() => AssertQuerySuccess("{ inputWithDefault(arg: null) }", @"{ ""inputWithDefault"": ""Grumpy"" }");
```

Comment:
I think I understand the logic, but it seems to me contradictory (at least for value types now). We now distinguish between cases when the argument is specified in request as `null` explicitly and when it is not specified at all (so default value will be used either from variable or field definition). In case of nullable enumeration graphtype for input I expect that `context.GetArgument<MyEnum>("arg")` **should** throw if `null` was explicitly specified instead of return default value (0/Grumpy in this case). Note that if your argument is required (`NonNullGraphType<EnumerationGraphType<MyEnum>>` then you free to call `context.GetArgument<MyEnum>("arg")` and **always** expect some valid value. But if your argument defined as `EnumerationGraphType<MyEnum>` (optional) then it will be more correct to call `context.GetArgument<MyEnum?>("arg")`. In my opinion this is logical.

This also applies to all value types generic parameters passed to `GetArgument`.

Also please add test for `{ inputRequired(arg: null) }` and `{ inputRequiredWithDefault(arg: null) }`

Reply:
Well... Now I see that `GetArgument` has additional ` TType defaultValue = default` parameter. Then that explains everything. Ok. 

So with provided `arg: null`
`context.GetArgument<MyEnum>("arg")` should yield `(MyEnum)0`
and
`context.GetArgument<MyEnum?>("arg")` should yield `null`

Do you agree? If so, let's add tests for nullable enums (and probably for other nullable value types - `int?`, `bool?`).

Reply:
Exactly.  Here with the enum tests, I'll add one with nullable equals `null` and change the existing one to say `(MyEnum)0` so the logic is clearer.  I added a whole bunch of tests for nullability in another file.

Reply:
Added a comment and a test